src/contract_wrappers/modules/sto/__tests__/usd_tiered_sto_wrapper.test.ts

```diff
@@ -367,7 +276,7 @@ describe('USDTieredSTOWrapper', () => {
       // Mock getRate
       const expectedGetRateResult = new BigNumber(1);
       // Mocked method
-      const mockedGetRateMethod = mock(MockedCallMethod);
+      const mockedGetRateMethod = mock(MockedSendMethod);
```

Comment:
- Previously these functions were getters that only had callAsync included in their wrappers.

Now, they access an oracle so the abi-wrapper doesnt seem them as just callAsync, now they are also sendTransaction. Based on the generated contract methods I have to use the mockedCall or mockedSend method type.

I as well disagreed with this at first, and asked in one case why the team didn't make the function *view* in solidity.

Adam Response to one of these cases.

> Adam Dossa  [15 days ago]

> It's hard to make it a view function as it may call out to an Oracle (if the price is denominated in USD and needs to be turned into POLY) which may involve state updates.

> You can still call the function as if it were a view function (i.e. without committing a tx to the chain) though and get back a response.

So I changed to mockedsendMethod so that it would allow test project to build

Reply:
But Adam states in his response that you can still just use `call` on the method. Why would it not build?

Reply:
Because the wrapper generated it in a sendTransactionAsync format and not a callAsync format. The code is still in the configuration of a call method, but in order to satisfy the compilation needs to use sendTransaction method. For me, the only workaround is changing the wrappers by modifying solidity, which we cannot do.  getRate must be a mockedsendmethod, but we only use its callAsync part.

![Screenshot (15)](https://user-images.githubusercontent.com/35141155/61137912-4827a080-a47b-11e9-89d2-05e643558343.png)

Reply:
I would have preferred to keep it as mockedcallmethod but I dont know what other option is available

Reply:
Oh, I see. I misunderstood the issue.

It's ok this way. If it's a limitation of the contracts there's nothing we can do about it.